### ISSUE TITLE
Remove `hupper` from debug server

### ISSUE THREAD

**Issue by patrick91** [strawberry-graphql/strawberry#1108]
Looks like uvicorn has a reload option that we can use instead of building our own reload mechanism: https://github.com/encode/uvicorn/blob/b5af1049e63c059dc750a450c807b9768f91e906/uvicorn/main.py#L410-L431

**Comment by DoctorJohn** [comment]
Nice find! Unfortunately using it isn't as straightforward as I expected. The uvicorn reload feature requires us to pass the ASGI application as an import string. This makes it harder to dynamically configure the ASGI app with the schema the user provided via the `strawberry server` command.

I read some of the uvicorn code and it appears that there is no way around passing the ASGI app as an import string. However, I discovered that the module containing the ASGI app still has access to the CLI arguments originally supplied by the user. These can be parsed again to obtain the schema import string which then can be used to import the schema and to finnaly configure the ASGI app.

I just tested this approche and it appears to be working quite well. I'll do some extra testing and create a PR. We then can ponder about whether this approche is to hacky or not :D

**Comment by patrick91** [comment]
oh, nice one! do you think we could improve uvicorn to make this easier for us?

**Comment by DoctorJohn** [comment]
> do you think we could improve uvicorn to make this easier for us?

I thought about it but the way there reloader works there isn't much one can do. But I guess thats fine.

I did some extra testing with the CLI argument based approche outlined before. While the approach itself worked it felt too error-prone to me. I ended up using an environment variable to pass the schema import string supplied by the user via the CLI to the asgi app. This approche works very well and feels less hacky.
